pkg/synchromanager/clustersynchro_manager.go

```diff
@@ -286,6 +312,10 @@ func (manager *Manager) reconcileCluster(cluster *clusterv1alpha2.PediaCluster)
 		}
 	}
 
+	if cluster.Spec.ShardingName != manager.shardingName {
+		return manager.reconcileShardingUnmanaged(cluster)
+	}
+
```

Comment:
We can set `status.ShardingName` to `manager.shardingName` in time here, and update it to indicate that the PediaCluster has been worked on by this manager before the manager updates the rest of the PediaCluster's status. Otherwise, the shardingName might stay nil, but the other status might keep updating due to an anomaly